### PR TITLE
Properly cross sign tokens

### LINKED ISSUE #114: Auth can not sign token using ECDS key
unexpected error: "key is invalid"

### PR DESCRIPTION
**Pull request by jbygdell**
* Only transfer the payload data to the new token
* Fixes signing with a ECDSA key
* Add test for signing with ECDSA as well ass RSA
* Update to GO 1.17
* Update linter version used in CI

Fixes #114

### CODE REVIEW

**Review by dbampalikis — APPROVED** [review]
Looks good and it seems to work on the cluster

File: elixir.go [hunk 1]
```diff
@@ -126,24 +129,31 @@ func generateJwtFromElixir(idStruct ElixirIdentity, key, alg string) (string, er
 		log.Error("Claims in token are empty")
 	}
 
+	u, err := url.Parse(iss)
+		if err != nil {
+		return "", fmt.Errorf("failed to pars ISS, reason: %v", err)
```
Comment (norling) [inline elixir.go:134]: there was a typo, so I made the message clearer for those of us who always forget the fields ^^
```suggestion
	return "", fmt.Errorf("failed to parse ISS (jwt issuer field), reason: %v", err)
```

**Review by norling — APPROVED** [review]

**Comment by jbygdell** [comment]
> Looks good and it seems to work on the cluster

This is not tested on the cluster, I added a RSA key there just to confirm my suspicions.

**Comment by dbampalikis** [comment]
> > Looks good and it seems to work on the cluster
> 
> This is not tested on the cluster, I added a RSA key there just to confirm my suspicions.

The auth seems to work now on the login.bp.nbis.se. Is that unrelated?

**Comment by jbygdell** [comment]
> > > Looks good and it seems to work on the cluster
> > 
> > 
> > This is not tested on the cluster, I added a RSA key there just to confirm my suspicions.
> 
> The auth seems to work now on the login.bp.nbis.se. Is that unrelated?

Nope, This PR fixes the original reason the login didn't work i.e. cross-signing the token with a different key type

**Review by aaperis — APPROVED** [review]
looks good :-)

**Pull request merged**